### PR TITLE
Mesh 3 : small improvements

### PR DESCRIPTION
**Pull request by janetournois**
## Summary of Changes
This PR introduces small improvements that we have recently made to the `Mesh_3` package, ranging from small bug fixes to time and memory footprint improvement.

## Release Management

* Affected package : Mesh_3

### CODE REVIEW

File: Mesh_3/include/CGAL/Mesh_3/Refine_facets_3.h [hunk 1]
```diff
@@ -483,7 +483,7 @@ class Refine_facets_3_base
   bool is_encroached_facet_refinable(Facet& facet) const;
 
   /// Insert facet into refinement queue
-  void insert_bad_facet(Facet facet, const Quality& quality)
+  void insert_bad_facet(Facet& facet, const Quality& quality)
```
Comment (lrineau) [inline Mesh_3/include/CGAL/Mesh_3/Refine_facets_3.h]: Do you know why this was changed?

Comment (janetournois) [inline Mesh_3/include/CGAL/Mesh_3/Refine_facets_3.h]: Thank you for asking!
The reverse modification was made by you in the commit c55e391f97eba43e93a2b9232b57f69c1a08174a, in the context of making the manifold criterion available with Mesh_3 parallel (PR #1911).
I must have mis-synchronized our private branch and this one. I'm fixing it!

Comment (janetournois) [inline Mesh_3/include/CGAL/Mesh_3/Refine_facets_3.h]: fixed by commit  	706b53b

**Comment by lrineau** [comment]
I just pushed the recent modification to Mesh_3 in this PR: https://github.com/CGAL/cgal/pull/2072/files/706b53b2c278780b0434cf563a62c4d8ef1b7527..b0317354e748c42b0acaa4ee7f4c6da4d0445877

**Comment by lrineau** [comment]
I did merge that PR on Monday (31f277dc93ce0b60cadab8248e999233f6d66699), but the commit 81dccb7d2290be5b78e3be31c4a3efdf580cef7a (that adds the debugging macro `CGAL_COMPACT_CONTAINER_DEBUG_TIME_STAMP`) actually triggers segfaults, with recent clang versions and optimizers (at least `-O2`, because `-O1` does not trigger the segfault).

I am working on it...

Cc @janetournois @afabri...

**Comment by lrineau** [comment]
I evaluate a solution , with commit b9681f48d3aaa947077f4485626ecce987988194...

**Pull request merged**